src/microbe_stage/systems/MicrobeAISystem.cs

```diff
@@ -510,6 +533,82 @@ private void ChooseActions(in Entity entity, ref MicrobeAI ai, ref CompoundAbsor
         }
     }
 
+    private (Entity Entity, Vector3 Position, CompoundBag Compounds)? GetNearestRadioactiveChunk(
+        ref WorldPosition position, float maxDistance)
+    {
+        (Entity Entity, Vector3 Position, CompoundBag Compounds)? chosenChunk = null;
+        float bestFoundChunkDistance = float.MaxValue;
+
+        BuildChunksCache();
+
+        foreach (var chunk in terrainChunkDataCache)
+        {
+            if (!chunk.Compounds.Compounds.Keys.Contains(Compound.Radiation))
+            {
+                continue;
+            }
+
+            var distance = (chunk.Position - position.Position).LengthSquared();
+
+            if (distance > bestFoundChunkDistance)
+                continue;
+
+            if (distance > maxDistance)
+                continue;
+
+            chosenChunk = chunk;
+        }
+
+        return chosenChunk;
+    }
+
+    private bool RunFromNearestRadioactiveChunk(ref WorldPosition position, ref MicrobeAI ai,
+        ref MicrobeControl control)
+    {
+        var chosenChunk = GetNearestRadioactiveChunk(ref position, 500.0f);
+
+        if (chosenChunk == null)
+        {
+            return false;
+        }
+
+        var oppositeDirection = position.Position + (position.Position - chosenChunk.Value.Position);
+
+        ai.TargetPosition = oppositeDirection;
+        control.LookAtPoint = ai.TargetPosition;
+
+        control.SetMoveSpeedTowardsPoint(ref position, ai.TargetPosition, Constants.AI_BASE_MOVEMENT);
+        control.Sprinting = true;
+
+        return true;
+    }
+
+    private bool GoNearRadioactiveChunk(ref WorldPosition position, ref MicrobeAI ai,
+        ref MicrobeControl control, float speciesFocus)
+    {
+        var maxDistance = 30000.0f * speciesFocus / Constants.MAX_SPECIES_FOCUS + 3000.0f;
+        var chosenChunk = GetNearestRadioactiveChunk(ref position, maxDistance);
+
+        if (chosenChunk == null)
+        {
+            return false;
+        }
+
+        // If the microbe is close to the chunk it doesn't need to go any closer
+        if (position.Position.DistanceSquaredTo(chosenChunk.Value.Position) < 700.0f)
```

Comment:
Maybe a slight bit of randomness in the distance could be used here to make cells not always stop at an exact distance away from a chunk? (the microbe AI should already have access to a random instance that is fetched for each entity)